File: dmn-test-cases/standard/tck/1.1/cl3/0013-sort/translator/expected/java/dmn/type/TRowImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tRow"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TRowImpl implements TRow {
        private java.lang.Number col1;
        private java.lang.Number col2;
        private java.lang.Number col3;
        private java.lang.Number col4;

    public TRowImpl() {
    }

    public TRowImpl(java.lang.Number col1, java.lang.Number col2, java.lang.Number col3, java.lang.Number col4) {
        this.setCol1(col1);
        this.setCol2(col2);
        this.setCol3(col3);
        this.setCol4(col4);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("col1")
    public java.lang.Number getCol1() {
        return this.col1;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("col1")
    public void setCol1(java.lang.Number col1) {
        this.col1 = col1;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("col2")
    public java.lang.Number getCol2() {
        return this.col2;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("col2")
    public void setCol2(java.lang.Number col2) {
        this.col2 = col2;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("col3")
    public java.lang.Number getCol3() {
        return this.col3;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("col3")
    public void setCol3(java.lang.Number col3) {
        this.col3 = col3;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("col4")
    public java.lang.Number getCol4() {
        return this.col4;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("col4")
    public void setCol4(java.lang.Number col4) {
        this.col4 = col4;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0013-sort/translator/expected/kotlin/dmn/type/TRowImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tRow"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TRowImpl : TRow {
    @get:com.fasterxml.jackson.annotation.JsonGetter("col1")
    @set:com.fasterxml.jackson.annotation.JsonGetter("col1")
    override var col1: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("col2")
    @set:com.fasterxml.jackson.annotation.JsonGetter("col2")
    override var col2: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("col3")
    @set:com.fasterxml.jackson.annotation.JsonGetter("col3")
    override var col3: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("col4")
    @set:com.fasterxml.jackson.annotation.JsonGetter("col4")
    override var col4: kotlin.Number? = null

    constructor() {
    }

    constructor (col1: kotlin.Number?, col2: kotlin.Number?, col3: kotlin.Number?, col4: kotlin.Number?) {
        this.col1 = col1
        this.col2 = col2
        this.col3 = col3
        this.col4 = col4
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
